### PR TITLE
Validate GTFS route type

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Requires https://github.com/conveyal/gtfs-lib/pull/343.

This PR adds validation for the `route_type` in the `routes` table on upload/fetching of a GTFS feed version.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]

File: src/main/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilder.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                        options.add(valueNode.intValue());
+                    }
+                }
+            }
+        }
+        return options;
+    }
+
+    /**
+     * Helper method to get a node with the given key and value
```
Comment (br648) [inline src/main/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilder.java]: Add a full stop at the end of the comment.

File: src/test/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class RouteTypeValidatorBuilderTest extends UnitTest {
+    @BeforeAll
+    static void setUp() throws IOException {
+        // start server if it isn't already running
```
Comment (br648) [inline src/test/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilderTest.java]: Full stop.

**Review by br648 — APPROVED** [review]
Two very minor comments, apart from that all good.

File: src/main/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilder.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilder.java]: Thanks, updated in https://github.com/ibi-group/datatools-server/pull/453/commits/c2d0ec01d7f9e1e6a4d54c08429d79ae6b06317a.

File: src/test/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilderTest.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/test/java/com/conveyal/datatools/manager/jobs/validation/RouteTypeValidatorBuilderTest.java]: Updated in https://github.com/ibi-group/datatools-server/pull/453/commits/c2d0ec01d7f9e1e6a4d54c08429d79ae6b06317a.

**Pull request merged**